### PR TITLE
Bugfix FXIOS-9283 [Reader Mode] fixed reader mode access

### PR DESCRIPTION
**Pull request by tusharC95**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9283)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/20561)

## :bulb: Description
Added `ReaderModeState` check when the tab's current title and last title are same.
## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift [hunk 1]
```diff
@@ -1697,9 +1697,17 @@ class BrowserViewController: UIViewController,
...
+            if !title.isEmpty {
+                if title != tab.lastTitle {
+                    tab.lastTitle = title
+                    navigateInTab(tab: tab, webViewStatus: .title)
+                } else {
+                    if let readerMode = tab.getContentScript(name: ReaderMode.name()) as? ReaderMode {
+                        if readerMode.state == .active {
+                            navigateInTab(tab: tab, webViewStatus: .title)
+                        }
+                    }
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:1709]: nit: what do you think about moving this into a private func with a descriptive name that gives us a better clue on why we need to add this logic.

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift [hunk 2]
```diff
@@ -1697,9 +1697,17 @@ class BrowserViewController: UIViewController,
...
+                }
             }
             TelemetryWrapper.recordEvent(category: .action, method: .navigate, object: .tab)
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:1712]: this seems like it should be called when calling the navigate tab, but separate from this ticket so we can leave as is.

File: firefox-ios/firefox-ios-tests/Tests/ClientTests/BrowserViewControllerTests.swift [hunk 3]
```diff
@@ -65,4 +65,36 @@ class BrowserViewControllerTests: XCTestCase {
 
         wait(for: [expectation], timeout: 5.0)
     }
+
+    func testTitleChangeEnablesReaderMode() {
+        guard let mockTab = loadMockTabWithTitle(title: "Webview title"), let tabWebView = mockTab.webView else { return }
```
Comment (cyndichin) [inline firefox-ios/firefox-ios-tests/Tests/ClientTests/BrowserViewControllerTests.swift:70]: seems this test is failing it bitrise

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift (hunk 1 above)
Comment (tusharC95) [reply firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:1709]: Yes, I will shift it to a private function.

File: firefox-ios/firefox-ios-tests/Tests/ClientTests/BrowserViewControllerTests.swift (hunk 3 above)
Comment (tusharC95) [reply firefox-ios/firefox-ios-tests/Tests/ClientTests/BrowserViewControllerTests.swift:70]: It is passing the test when tested repeatedly.  Seems like asynchronous wait is failing in Bitrise. I will update the wait time to see if the test case passes.

**Review by cyndichin — APPROVED** [review]
looks good! seems like unit tests are a bit difficult to add with our current state so discussed that we will look into UI tests (separate from this work). thanks for the great work!

**Pull request merged**